### DISCUSSION TITLE
How to calculate Fuzz to crop based on the image border?

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#3942] [Digital Image Processing]
convert --version
Version: ImageMagick 6.9.10-71 Q16 x86_64 2021-07-06 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC OpenMP(3.1) 
Delegates (built-in): bzlib fftw fontconfig freetype jng jp2 jpeg lzma openexr pangocairo png tiff x xml zlib

I have 2 images.

![3](https://user-images.githubusercontent.com/12046302/125880294-779f9994-0854-47e7-bba1-a282668a78dd.jpg)
![23](https://user-images.githubusercontent.com/12046302/125880306-2eefaa38-7daa-4c93-8858-7f59da57d283.jpg)

I am using this from a old post, to crop the border

magick "$FILE" -set option:color "%[pixel:u.p{10,10}]" -background "%[color]" -deskew 90% +repage \
\( -clone 0 -fuzz 22%% -fill black -draw "color 10,10 floodfill" -alpha off +fuzz -fill white +opaque black -type bilevel \
-define connected-components:mean-color=true \
-define connected-components:area-threshold=1000 \
-connected-components 8 -set option:cropbox "%@" +delete \) \
-crop "%[cropbox]" +repage -evaluate Pow %[fx:log\(0.6\)/log\(mean\)] -unsharp 0x5 \
"$FILENAME"-borderoff.png

The Fuzz value of 22% is working on the first one.

![3-borderoff](https://user-images.githubusercontent.com/12046302/125880594-a2d5820f-0e71-477e-b7cf-d07a0704c3b4.png)

Fuzz value 8% on the other one. Otherwise, even the text is cropped.

![23-borderoff](https://user-images.githubusercontent.com/12046302/125880583-aaa86a1f-7e07-453a-a571-4eee98a1006b.png)

Thanks,

**Comment by snibgo** [comment]
> The Fuzz value of 22% is working on the first one.

But it removes text. Do you want to remove the text?

**Reply by srini8080** [reply]
no. I want to just crop the borders off and cleanup the page to have white background, colors intact.

Also when I use greater fuzz, the text is getting blurred. I want the text sharp enough to read.

Thanks

**Comment by fmw42** [comment]
My suggestion would be to compute what is needed to trim all the border off. Then recompute adding back some pad. Then crop.

![img](https://user-images.githubusercontent.com/13458218/125893324-d39c0f1f-b65c-4981-988d-e326c3126a03.jpg)

Unix bash scripting

```
thick=10
trimbox=`convert img.jpg -fuzz 10% -format "%@" info: | tr "x" "+"`
wd=`echo $trimbox | cut -d+ -f1`
ht=`echo $trimbox | cut -d+ -f2`
xo=`echo $trimbox | cut -d+ -f3`
yo=`echo $trimbox | cut -d+ -f4`
wd=$((wd+2*thick))
ht=$((ht+2*thick))
xo=$((xo-thick))
yo=$((yo-thick))
trimbox="${wd}x${ht}+${xo}+${y0}"
convert img.jpg -crop $trimbox +repage result.png

```

![result](https://user-images.githubusercontent.com/13458218/125893374-26482e56-2e67-46d0-bbb7-497897502f0a.png)


Adjust the fuzz as needed to trim as desired. Then add as much pad (thick) back. Enhance as desired.

**Reply by srini8080** [reply]
The challenge has been, there are over 100 pages. Some are well scanned, some are very fuzzy. How do we adjust fuzz based on the page?

**Reply by fmw42** [reply]
There is no universal solution for fuzz! That is why I fuzzed to trim to bounding box (but not too much, so that it was not too sensitive). Fuzz so that text is not trimmed but all else is.

**Reply by srini8080** [reply]
Is there something in the image that I can use to say that the fuzz should be more or less, then we can do some guess around it

Thanks

**Reply by fmw42** [reply]
Nothing that I know that works every time or even most of the time.

**Reply by srini8080** [reply]
Also, what is the best way to make the background of these pages white?

Thanks

**Reply by srini8080** [reply]
basically because of the noise around the image, the cropping is not working well.

I tried to tone it also using the autotone script.

Thanks.
![9-borderoff2](https://user-images.githubusercontent.com/12046302/125905863-c3436600-50c5-4512-ae35-c8263c0c5450.png)
